$NetBSD: patch-src_libmongoc_CMakeLists.txt,v 1.1 2026/08/03 15:24:32 wiz Exp $

Fix build with cmake 4.4.
://github.com/mongodb/mongo-c-driver/commit/ea1b54791ec2f1aafb665f98af6cc9dd21116782

--- src/libmongoc/CMakeLists.txt.orig	2024-02-15 16:27:12.000000000 +0000
+++ src/libmongoc/CMakeLists.txt
@@ -430,9 +430,12 @@ function (mongoc_get_accept_args ARG2 ARG3)
    }
    ")
 
-   TRY_COMPILE (RES ${CMAKE_CURRENT_BINARY_DIR}
-   ${CMAKE_CURRENT_BINARY_DIR}/accept_test${VAR}.c CMAKE_FLAGS
-   "-Werror -DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'" OUTPUT_VARIABLE LOG2)
+   TRY_COMPILE (RES
+      "${CMAKE_CURRENT_BINARY_DIR}"
+      SOURCES "${CMAKE_CURRENT_BINARY_DIR}/accept_test${VAR}.c"
+      CMAKE_FLAGS "-DCOMPILE_DEFINITIONS=-Werror" "-DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'"
+      OUTPUT_VARIABLE LOG2
+   )
 
    if (RES)
       message (
